### PR TITLE
Frontend Buildcheck Workflow Revamp

### PR DESCRIPTION
**Pull request by prishavall**
# Frontend Buildcheck Workflow Revamp
#1086 
**What user problem are we solving?**
Creating a unified Frontend build check workflow to ensure code quality and consistency
**What solution does this PR provide?**
This creates the build check along with pnpm features and a specific working directory point to the frontend
**Testing Methodology**
> How did you test your changes and verify that existing functionality is not broken
Tested using vscode and general tests.
**Any other considerations**

### CODE REVIEW

**Comment by farisdurrani** [comment]
Please fix build errors @prishavall

File: .github/workflows/frontend.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+concurrency: preview-${{ github.ref }}
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (karkir0003) [inline .github/workflows/frontend.yml]: I think `actions/checkout@v3` is most up to date. can you check other `.github/workflow` yaml files

File: .github/workflows/frontend.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Run ESLint
+        run: pnpm run lint
+        working-directory: ./frontend
+
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (karkir0003) [inline .github/workflows/frontend.yml]: might need to update the version of `actions/checkout` to `actions/checkout@v3`. confirm with other yaml files in `.github/workflows` directory

File: frontend/package.json [hunk 3]
```diff
@@ -4,7 +4,8 @@
   "scripts": {
     "preinstall": "npx only-allow pnpm",
     "lint": "next lint",
-    "start": "next dev"
+    "start": "next dev",
+    "build": "next build"
```
Comment (karkir0003) [inline frontend/package.json]: remove this one for now since we aren't using it

Comment (karkir0003) [reply frontend/package.json]: I mean the `"build": "next build"` line

**Review by farisdurrani — APPROVED** [review]

**Comment by farisdurrani** [comment]
Apologies for the delay, @prishavall . I'm okay with the changes and I'll merge this in for you

**Pull request merged**